sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql

```diff
@@ -0,0 +1,222 @@
+    sample_id,
+    submission_timestamp,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_ad_group, "") AS adjust_ad_group,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_campaign, "") AS adjust_campaign,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_creative, "") AS adjust_creative,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_network, "") AS adjust_network,
+  FROM
+    fenix.metrics AS fenix_metrics
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND client_info.client_id IS NOT NULL
+),
+metrics_ping AS (
+  SELECT
+    client_id,
+    sample_id,
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp ASC
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    metrics_ping_base
+  GROUP BY
+    client_id,
+    sample_id
+),
+_current AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+    first_reported_country,
+    first_reported_isp,
+    channel,
+    device_manufacturer,
+    device_model,
+    os_version,
+    app_version,
+    COALESCE(first_session.adjust_info, metrics.adjust_info) AS adjust_info,
+    STRUCT(
+      IF(first_session.client_id IS NULL, FALSE, TRUE) AS is_reported_first_session_ping,
+      IF(metrics.client_id IS NULL, FALSE, TRUE) AS is_reported_metrics_ping,
+      CASE
+        WHEN first_session.adjust_info IS NOT NULL
+          THEN "first_session"
+        WHEN metrics.adjust_info IS NOT NULL
+          THEN "metrics"
+        ELSE NULL
+      END AS adjust_info__source_ping
+    ) AS metadata,
+  FROM
+    first_seen
+  FULL OUTER JOIN
+    first_session_ping AS first_session
+  USING
+    (client_id, sample_id)
+  FULL OUTER JOIN
+    metrics_ping AS metrics
+  USING
+    (client_id, sample_id)
+  WHERE
+    client_id IS NOT NULL
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    fenix_derived.firefox_android_clients_v2
+)
+SELECT
+  client_id,
+  sample_id,
+  COALESCE(_previous.first_seen_date, _current.first_seen_date) AS first_seen_date,
+  COALESCE(
+    _previous.first_reported_country,
+    _current.first_reported_country
+  ) AS first_reported_country,
+  COALESCE(_previous.first_reported_isp, _current.first_reported_isp) AS first_reported_isp,
+  COALESCE(_previous.channel, _current.channel) AS channel,
+  COALESCE(_previous.device_manufacturer, _current.device_manufacturer) AS device_manufacturer,
+  COALESCE(_previous.device_model, _current.device_model) AS device_model,
+  COALESCE(_previous.os_version, _current.os_version) AS os_version,
+  COALESCE(_previous.app_version, _current.app_version) AS app_version,
```

Comment:
Can you confirm this isn't missing any fields from `firefox_android_clients`? (sans `last_reported` fields, we can drop those). I'm seeing e.g. locale missing.

Reply:
As far as I can see they are all there, locale should be there now too.